lib/rouge/lexers/email.rb

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Email < RegexLexer
+      tag 'email'
+      aliases 'eml', 'mail'
+      filenames '*.eml', '*.mail'
+      mimetypes 'message/rfc822'
+
+      title "Email"
+      desc "An email message"
+
+      def self.detect?(text)
+        return true if text.start_with?('From: ')
+      end
+
+      start do
+        push :header
+      end
+
+      state :header do
+        rule %r/^(From|To|Cc|Bcc):\s/, Keyword, :address
+        rule %r/^Date:\s/, Keyword, :date
+        rule %r/^Subject:\s/, Keyword, :subject
+        rule %r/^[A-Z][A-Za-z0-9-]+:\s/, Keyword, :other_header
+        rule %r/\n/m, Text::Whitespace, :pop!
```

Comment:
Just a note: the `/m` here doesn't actually do anything in Ruby - it only changes the behaviour of `/./`, which normally doesn't match newlines. Matching against an explicit `\n` is always "multiline" :]

Reply:
see https://ruby-doc.org/core-2.5.1/Regexp.html#class-Regexp-label-Options

Reply:
Ah, thanks.  I removed them.